CRM/Contribute/BAO/Contribution.php

```diff
@@ -1261,18 +1262,10 @@ public static function deleteContribution($id) {
     if (CRM_Price_BAO_PriceSet::getFor('civicrm_contribution', $id)) {
       CRM_Price_BAO_PriceSet::removeFrom('civicrm_contribution', $id);
     }
-    // cleanup line items.
-    $participantId = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_ParticipantPayment', $id, 'participant_id', 'contribution_id');
 
-    // delete any related entity_financial_trxn, financial_trxn and financial_item records.
+    // delete any related financial records.
     CRM_Core_BAO_FinancialTrxn::deleteFinancialTrxn($id);
-
-    if ($participantId) {
-      CRM_Price_BAO_LineItem::deleteLineItems($participantId, 'civicrm_participant');
-    }
-    else {
-      CRM_Price_BAO_LineItem::deleteLineItems($id, 'civicrm_contribution');
-    }
+    LineItem::delete(FALSE)->addWhere('contribution_id', '=', $id)->execute();
 
```

Comment:
looks good to me .. I was checking what else `deleteLineItems` could do and there's nothing additional. And eventually, we can get rid of that function in other places: 
```
CRM/Member/BAO/Membership.php:379:          CRM_Price_BAO_LineItem::deleteLineItems($ids['membership'], 'civicrm_membership');
CRM/Member/BAO/Membership.php:672:    CRM_Price_BAO_LineItem::deleteLineItems($membershipId, 'civicrm_membership');
CRM/Price/BAO/LineItem.php:348:  public static function deleteLineItems($entityId, $entityTable) {
CRM/Contribute/Form/Contribution.php:1525:        CRM_Price_BAO_LineItem::deleteLineItems($this->_id, 'civicrm_contribution');
CRM/Event/Form/Registration/Confirm.php:651:            CRM_Price_BAO_LineItem::deleteLineItems($entityId, $entityTable);
CRM/Event/Form/Participant.php:1113:      CRM_Price_BAO_LineItem::deleteLineItems($this->_id, 'civicrm_participant');
CRM/Event/BAO/Participant.php:858:    CRM_Price_BAO_LineItem::deleteLineItems($participantsId, 'civicrm_participant');
```

Reply:
@monishdeb yep - all the other places are scary too....